seed/lib/xml_mapping/reader.py

```diff
@@ -0,0 +1,39 @@
+# !/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2019, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+
+import os
+import zipfile
+
+
+class BuildingSyncParser(object):
+    def __init__(self, file_):
+        """
+        :param file_: FieldFile, an ImportFile's file
+        """
+
+        filename = file_.name
+        _, file_extension = os.path.splitext(filename)
+
+        # grab the data from the zip or xml file
+        self.data = []
+        if file_extension == '.zip':
+            with zipfile.ZipFile(file_, 'r', zipfile.ZIP_STORED) as openzip:
+                filelist = openzip.infolist()
+                for f in filelist:
+                    if '.xml' in f.filename and '__MACOSX' not in f.filename:
+                        self.data.append({'_xml': openzip.read(f).decode(), '_filename': f.filename})
```

Comment:
@adrian-lara @nllong 
Looking for feedback on this section of code. Currently it's reading _all files_ into memory (yikes) to represent a 'row' of data (ie property instance). I'm looking for better ideas for what to do here. These are the two options I can think of:
1. turn self.data into a generator within the context of the zip file. Not sure what the implications are here.
2. Read from the zip for every property when mapping. This would require adding a field to each data entry, '_zip', where when true indicates it's from a zip file. Then when we go to read/map the data, we know we need to open the import_file.file to get the actual file info. I'm concerned about concurrent reads from the workers, but it should probably be fine.

Reply:
After initial research I've decided to just stick with reading the files into memory, as is currently implemented. In the future will will probably want to implement a generator class for the `data` property.